spices/SPICE-0011-const-object-members.adoc

```diff
@@ -0,0 +1,111 @@
+= Const Object Members
+
+* Proposal: link:./SPICE-0011-const-object-members.adoc[SPICE-0011]
+* Author: https://github.com/bioball[Dan Chao]
+* Status: Review
+* Implemented in: TBD
+* Category: Language
+
+== Introduction
+
+This SPICE proposes allowing object members to be `const`.
+
+== Motivation
+
+Today, the `const` modifier can only be applied to class properties.
+
+However, there are times where this rule is insufficient.
+
+When a module amends another module, it has special rules; top-level properties are considered _object properties_, and not _class properties_.
+
+This means that these properties cannot be declared `const`.
+
+[source,pkl]
+----
+amends "foo.pkl"
+
+const bar = 5 // Pkl Error: modifier `const` cannot be applied to object members
+----
+
+Amending modules can also contain classes and typealiases.
+These members must be declared with the `local` modifier.
+
+However, these members cannot reference their enclosing module's properties, because classes and typealiases can only reference properties that are `const`.
+
+One possible scenario is to define a test fixture.
+
+[source,pkl]
+----
+amends "pkl:test"
+
+import "birds.pkl"
+
+local warbler = "Warbler"
+
+local class Warbler extends birds.Bird {
+  name = warbler // Pkl Error: cannot reference `warbler` from here because it is not `const`
+}
+----
+
+There is no straightforward way to resolve this, because `warbler` cannot be declared `const`:
+
+[source,pkl]
+----
+amends "pkl:test"
+
+import "birds.pkl"
+
+local const warbler = "Warbler" // Pkl Error: modifier `const` cannot be applied to object members
+
+local class Warbler extends birds.Bird {
+  name = warbler
+}
+----
+
+== Proposed Solution
+
+To fix this papercut, a object members are allowed to be `const`, but must co-appear with the `local` modifier.
```

Comment:
```suggestion
To fix this papercut, object members are allowed to be `const`, but must also be `local`.
```

This phrasing feels very roundabout. One (somewhat pedantic) point is that _members_ not co-appearing with any of their modifiers, but rather modifier `const` requires modifier `local`. That said; `local` is what makes the property an object property to begin with.

If my understanding is correct, this phrasing is there to motivate the restriction of `const` to `local` object members only, i.e. to exclude `new Dynamic { const foo = 5 }`. It seems more direct to rephrase the proposal to be `const local members`, rather than `const object members`. Either way, it should be useful to this SPICE to include a description of the `Dynamic` case (which is the only other object property case I can think of).